src/Server/Program.cs

```diff
@@ -15,44 +15,78 @@
 using Microsoft.Extensions.DependencyInjection.Extensions;
 using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
-using Nvidia.Clara.Dicom.DicomWeb.Client;
-using Nvidia.Clara.Dicom.DicomWeb.Client.API;
 using Nvidia.Clara.DicomAdapter.API;
 using Nvidia.Clara.DicomAdapter.Common;
 using Nvidia.Clara.DicomAdapter.Configuration;
+using Nvidia.Clara.DicomAdapter.Database;
 using Nvidia.Clara.DicomAdapter.Server.Common;
 using Nvidia.Clara.DicomAdapter.Server.Repositories;
-using Nvidia.Clara.DicomAdapter.Server.Services.Config;
 using Nvidia.Clara.DicomAdapter.Server.Services.Disk;
 using Nvidia.Clara.DicomAdapter.Server.Services.Export;
 using Nvidia.Clara.DicomAdapter.Server.Services.Http;
 using Nvidia.Clara.DicomAdapter.Server.Services.Jobs;
 using Nvidia.Clara.DicomAdapter.Server.Services.Scp;
 using Nvidia.Clara.DicomAdapter.Server.Services.Scu;
-using Serilog;
 using System;
 using System.IO.Abstractions;
-using System.Threading.Tasks;
 
 namespace Nvidia.Clara.DicomAdapter
 {
-    public class Program : IAsyncDisposable
+    public class Program
     {
         private static readonly string ApplicationEntryDirectory = AppDomain.CurrentDomain.BaseDirectory;
-        private static Serilog.ILogger Logger;
-        private IHost _host;
 
-        public Program(string[] args)
+        private static void Main(string[] args)
         {
-            _host = new HostBuilder()
+            Console.WriteLine("Reading logging.config from {0}", ApplicationEntryDirectory);
+            Environment.CurrentDirectory = ApplicationEntryDirectory;
+            Environment.SetEnvironmentVariable("HOSTNAME", Environment.MachineName);
+            // ConfigureLogging();
+
+            var host = CreateHostBuilder(args).Build();
+            var loggerFactory = InitializeLogger(host);
+            LoadPlugins(loggerFactory.CreateLogger("PlugInLoader"));
+            InitializeDatabase(host);
+            host.Run();
+        }
+
+        private static void LoadPlugins(ILogger logger)
+        {
+            try
+            {
+                PlugInLoader.LoadExternalProcessors(logger);
+            }
+            catch (System.Exception ex)
+            {
+                logger.Log(Microsoft.Extensions.Logging.LogLevel.Error, ex, "Error loading plugins.");
+            }
+        }
+
+        private static ILoggerFactory InitializeLogger(IHost host)
+        {
+            var loggerFactory = new LoggerFactory();
+            return loggerFactory;
+        }
+
+        private static void InitializeDatabase(IHost host)
+        {
+            using (var serviceScope = host.Services.CreateScope())
+            {
+                var context = serviceScope.ServiceProvider.GetRequiredService<DicomAdapterContext>();
+                context.Database.Migrate();
+            }
+        }
+
+        public static IHostBuilder CreateHostBuilder(string[] args) =>
```

Comment:
ensure `args` are not `null` first?

Reply:
Commented above; can be handled if used inside of each `Configure...()` methods.